lib/serialisers/JSON06Serialiser.js

```diff
@@ -132,7 +135,13 @@ module.exports = class JSON06Serialiser extends JSONSerialiser {
       samples.unshift(element.content);
     }
 
-    samples = samples.map(sample => new this.namespace.elements.Array([sample.content]));
+    samples = samples.map((sample) => {
+      if (sample instanceof this.namespace.elements.Array) {
```

Comment:
Once I fixed the exception, I discovered the output from the serialiser to be incorrect, we was taking the members (sample.content) and wrapping them in an array. Instead I've added a check that if sample is already an array type (such as an object element) then we will just wrap the entire element in an array and not it's content (element).